### DISCUSSION TITLE
Problem with GetMiddleware

### DISCUSSION THREAD

**Discussion by liemfs** [jonataslaw/getx#1918] [Q&A]
I've just implemented GetMiddleware on my project. I thought it was good to validate the data right before the page was called, but I found a problem with my implementation.

### AppPage Routes:
```
static final routes = [
    GetPage(
      name: _Paths.PROFILE,
      page: () => ProfileView(),
      binding: ProfileBinding(),
      middlewares: [
        CheckAccountMiddleware(),
        CheckProfileMiddleware(),
      ],
      preventDuplicates: false,
    ),
  ];
```
### Code to call a page
```
Future<void> onNavigateToProfileViewTap() async {
    try {
      await Get.toNamed(Routes.PROFILE, arguments: {'profileId': account.phoneNumber}, parameters: {'profileId': account.phoneNumber});
    } catch (e) {
      _logger.log(log.Level.SEVERE, StackTrace.current.toString());
    }
  }
```
### CheckProfileMiddleware
```
class CheckProfileMiddleware extends GetMiddleware {
  final _basicService = Get.find<BasicService>();

  @override
  GetPage? onPageCalled(GetPage? page) {
    try {
      if (page == null) return null;

      print(page.arguments);
      print(page.parameters);

      print(Get.arguments);
      print(Get.parameters);

      return page;
    } catch (e) {
      _logger.log(log.Level.SEVERE, e);
    }
    return null;
  }
}
```
### Result
```
I/flutter ( 5176): null
I/flutter ( 5176): {profileId: xxxxxxxxxx}
I/flutter ( 5176): null
I/flutter ( 5176): {profileId: xxxxxxxxxx}
```

Arguments always return null
I can't use parameters because I need to parse object data as default value for the page.

Please help.

**Comment by csandovalgt** [comment]
I have the same issue... getting arguments from Middleware always return null. I'm using middleware to avoid users to access an specific route from a multistep form in which step x  needs arguments from step x - 1.

I also tried to make the redirect like this:

`
@override
  void initState() {
    final args = Get.arguments;
    if (args==null) {
      Get.toNamed("/otherRoute");
    }
    super.initState();
  }

`

but I get other errors because is rendering while setting State. Anybody has solve this? @jasonlaw

**Comment by Ali3Jamous** [comment]
Same issue until now..

I found that Get.arguments can only see the arguments you set in the Middlware itself.
If you add arguments from the Middlware code, the next call for this Middlware can see it. Looks like that the Middlwares has its own space in Get.
